tests/test_tpp_backend.py

```diff
@@ -5858,3 +5859,152 @@ def get_cursor(self):
+        age_at_first_visit=patients.with_an_ons_cis_record(
+            returning="age_at_visit", find_first_match_in_period=True
+        ),
+        # filter by a different date column; filters out patient 3
+        age_filtered_by_covid_test_pos=patients.with_an_ons_cis_record(
+            returning="age_at_visit",
+            date_filter_column="covid_test_blood_pos_first_date",
+            on_or_after="2021-06-15",
+            include_date_of_match=True,
+            date_format="YYYY-MM-DD",
+        ),
+        # filter by the returning value, if that value is a date itself and no date filter specified
+        covid_test_pos_date=patients.with_an_ons_cis_record(
+            returning="covid_test_blood_pos_first_date",
+            on_or_after="2021-06-15",
+            date_format="YYYY-MM-DD",
+        ),
+        # binary flag return value
+        has_visit_with_pos_test_before_2021_08=patients.with_an_ons_cis_record(
+            returning="binary_flag",
+            date_filter_column="covid_test_blood_pos_first_date",
+            on_or_before="2021-07-31",
+        ),
+        # number_of_matches_in_period return value
+        num_visits_with_pos_test_before_2021_08=patients.with_an_ons_cis_record(
+            returning="number_of_matches_in_period",
+            date_filter_column="covid_test_blood_pos_first_date",
+            on_or_before="2021-07-31",
+        ),
+        # boolean value
+        self_isolating_at_last_visit=patients.with_an_ons_cis_record(
+            returning="self_isolating"
+        ),
+        # Coded categories are converted to long-form string labels
+        # result_tdi is a coded category value; varchar type in the db
+        # raw values are stringified ints, return the long form category lable
+        result_tdi=patients.with_an_ons_cis_record(
+            returning="result_tdi",
+            find_first_match_in_period=True,
+            include_date_of_match=True,
+            date_format="YYYY-MM-DD",
+        ),
+        # country is a coded category value, int type in the db
+        # raw values are ints, return the long form category lable
+        country=patients.with_an_ons_cis_record(
+            returning="country",
+            find_first_match_in_period=True,
+            include_date_of_match=True,
+            date_format="YYYY-MM-DD",
+        ),
+        # return country as codes, with a date filter that filters out patient 3;
+        # Missing values returned as empty string
+        country_as_codes=patients.with_an_ons_cis_record(
+            returning="country",
+            return_category_labels=False,
+            find_first_match_in_period=True,
+            date_filter_column="covid_test_blood_pos_first_date",
+            on_or_after="2021-06-15",
+        ),
+    )
+
+    assert_results(
+        study.to_dicts(convert_to_strings=False),
+        age_at_visit=[20, 30, 41],
+        age_at_visit_date=["2021-10-01", "2021-10-01", "2021-10-01"],
```

Comment:
Thanks for returning the dates here (and elsewhere). It made associating the fixtures with the results much easier. It's also a helpful reminder that `visit_date` is the default date column.